src/utils/unitConvert.ts

```diff
@@ -37,13 +37,25 @@ export function fromSatsToUnitOrFractional(
     const unit = isLbtc ? lbtcUnit : undefined;
     return new Big(sats)
       .div(new Big(10).pow(new Big(precision).minus(unitToExponent(unit)).toNumber()))
-      .toString();
+      .toFixed();
```

Comment:
mmm, this round decimals to integer if too much 

ie. 5.56789512 become 6

Reply:
Say we have `fromUnitToUnit(5.56789512, 'L-bits', 'L-sats')`.
The amount is not correct because L-bits should be 2 decimal places max. What is the right approach? Should I truncate to '5.56' before converting?

Reply:
We could also throw since invalid numbers are not supposed to happen

Reply:
By sanitizing you mean having a function like `sanitizeLbtcAmount(amount: number, lbtcUnit)`.
If params are `5.56789512` and `L-bits` it truncate and returns `5.56` ? Or round up and returns `5.57` ?

Reply:
No, I think best to truncate the the number right where is generated: either via user entering the input or after a calcuation in stas, we should make sure there should never be a number more than the unit precision

Reply:
`toFixed()` is needed. I decided to return a string. If I remove it it returns Big type. `toString()` returns scientific notation in some cases.

Reply:
I feel safer with strings. Now if we decide to return number there is a bit a refacto to do

Reply:
Disagree re: safety. But let's move on for now and track for future refactors

https://github.com/tdex-network/tdex-dashboard/issues/304